prospector/autodetect.py

```diff
@@ -41,32 +42,29 @@ def find_from_imports(file_contents):
     return names
 
 
-def find_from_path(path):
+def find_from_path(path: Path):
     names = set()
 
     try:
-        dirlist = os.listdir(path)
+        for item in path.iterdir():
+            if item.is_dir():
+                if is_virtualenv(item):
+                    continue
+                names |= find_from_path(item)
+            elif not item.is_symlink() and item.suffix == ".py":
+                try:
+                    contents = encoding.read_py_file(item)
+                    names |= find_from_imports(contents)
+                except encoding.CouldNotHandleEncoding as err:
+                    # TODO: this output will break output formats such as JSON
+                    warnings.warn("{0}: {1}".format(err.path, err.__cause__), ImportWarning)
```

Comment:
The 0 and 1 have not been needed since Python 2.6.
* https://docs.python.org/3/whatsnew/2.6.html#pep-3101-advanced-string-formatting

Reply:
https://github.com/PyCQA/prospector/pull/542 😄